packages/jsii-spec/lib/spec.ts

```diff
@@ -409,7 +418,7 @@ export interface UnionTypeReference extends TypeReferenceBase {
          * All the possible types (including the primary type).
          * @minItems 2
          */
-        types: TypeReference[];
+        types: Array<TypeInstance<TypeReference>>;
```

Comment:
It does not make sense for a union type to be something like `number | string?` or `number | Promise<string>`.

Reply:
And yet, this is valid TypeScript, and thus could be authored in a TypeScript signature that JSII would totally parse out in that way

Reply:
Nobody says we can't reject that in jsii. Will the kernel handle it correctly?